### PR TITLE
✨ introduce search parameter to the listing workspaces

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- ✨ introduce search parameter to the listing workspaces

## Related issue/s
- closes https://github.com/ITISFoundation/osparc-simcore/issues/6871

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by odeimaiz — COMMENTED** [review]
👍

File: services/web/server/src/simcore_service_webserver/api/v0/openapi.yaml [hunk 1]
```diff
@@ -4540,7 +4540,7 @@ paths:
         '403':
           description: ProjectInvalidRightsError
         '404':
-          description: UserDefaultWalletNotFoundError, ProjectNotFoundError
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/api/v0/openapi.yaml:4543]: i think there is an issue with the generator. The order in this description is random and every time we run it, it results in a different order. Perhaps we can add some sorting?

File: services/web/server/src/simcore_service_webserver/workspaces/_models.py [hunk 2]
```diff
@@ -46,6 +48,14 @@ class WorkspacesFilters(Filters):
         default=False,
         description="Set to true to list trashed, false to list non-trashed (default), None to list all",
     )
+    text: Annotated[
+        str | None, BeforeValidator(empty_str_to_none_pre_validator)
+    ] = Field(
+        default=None,
+        description="Multi column full text search",
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/workspaces/_models.py:55]: which column? Shouldn you add that they are in `name` OR `description`?  Why don't you create a template message and use it in all `text` search?

File: services/web/server/src/simcore_service_webserver/workspaces/_models.py [hunk 3]
```diff
@@ -46,6 +48,14 @@ class WorkspacesFilters(Filters):
...
+        max_length=100,
+        examples=["My Workspace"],
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/workspaces/_models.py:57]: to be honest, an example like this does not add much. On the other hand, an example with `%` (or` *`) tokens might be interesting. Make sure to have the examples you have in your tests.

**Review by pcrespov — APPROVED** [review]
thx!

**Review by sanderegg — APPROVED** [review]
thanks

**Review by GitHK — APPROVED** [review]

**Review by bisgaard-itis — APPROVED** [review]
👍🏻

**Pull request merged**